### DISCUSSION TITLE
v10 testing

### DISCUSSION THREAD

**Discussion by stb68** [gitbls/sdm#143] [General]
Hi

I installed v10 over the current version and I've done some quick testing with the new mkdir, copydir and --plugin @

I should maybe add some of this as, or to existing, issues - if you'd like to move bits elsewhere or want me to add issues then let me know which for and I will do.

"user:group" still seems to not work and need to use "user.group" - although that now works without giving a warning.

**mkdir**: Seems perfect on my test.  Thank you

**copydir**:

Main thing with the copydir is if you want to pass some flags and something like --chmod=755 and/or --chown=USER:GROUP.  Because you can't pass spaces in the flag list, if you want to copy files other than as root (actually see later, that doesn't work either currently) or the user on the source system, or if you want to set permissions without having to worry about the source being as required, then you can't.

My initial thought for a simpler option, rather than "fixing" the parsing, was to maybe go with extra key options, e.g. rsyncopts, rsyncopts2 and rsynopts3 keys and tag them all onto the command?

Or maybe you'd prefer specific 'umask' and 'chown' keys directly handled by the plugin?  Or specific 'chown' and 'chmod' keys which then get mapped to '--chmod' and '--chown' on the actual rsync command?

There's a bigger issue though with the default -a flag if specifying extra flags.  I would argue that this needs to be removed if other flags are specified to stop clashes.

This paragraph started just by way of a note to maybe add somewhere, but is a good example of why the -a should be removable.  If copying into the /etc/skel area then it seems to be necessary to override the default -a so that the files belong to root.  Otherwise they don't seem to be copied when the user is created (at least I'm assuming that's why -> sequence of copydir '.config' as a user, copyfile '.bashrc' as root, create user and only the '.bashrc' is copied to the new user).

But, now I've tried with specifying -rpti, I still get the files belonging to the user number that they belonged to on the server that the burn was run on, rather than root.  I can see that the -a is still present in the underlying rsync call.

Something else I noticed before, the password for the created user (user created at burn time and password entered when prompted) being in the /etc/sdm/assets/user/redact-passwords file, even after the system has been setup and booted.  I'm not sure that's a good thing.

Typo in the hotspot plugin "Configure hotspot with hostpad and dhcpcd"  [sorry, ignore me :-)]

Using old style networking, I think there's extra config I need to do that I wouldn't on an off the shelf installation.  Particularly that eth0 doesn't get setup.  I didn't spot any instructions on this?

It seems that I need to enable / start the dhcpcd service and add some configuration for eth0.  I'm still working on that (don't get much time to try any of this, especially without many interruptions!).

Thanks for all the great work.

**Comment by gitbls** [comment]
> Hi
> 
> I installed v10 over the current version and I've done some quick testing with the new mkdir, copydir and --plugin @
> 
> I should maybe add some of this as, or to existing, issues - if you'd like to move bits elsewhere or want me to add issues then let me know which for and I will do.

Thanks for the feedback. This feedback format is fine...I prefer to focus on the content, rather than the "issue structure".
> 
> "user:group" still seems to not work and need to use "user.group" - although that now works without giving a warning.
> 

Can you provide a specific example where this doesn't work? I of course tested it before heave-ho, so how can I repro this?

> **mkdir**: Seems perfect on my test. Thank you
> 

Yay! 

> **copydir**:
> 
> Main thing with the copydir is if you want to pass some flags and something like --chmod=755 and/or --chown=USER:GROUP. Because you can't pass spaces in the flag list, if you want to copy files other than as root (actually see later, that doesn't work either currently) or the user on the source system, or if you want to set permissions without having to worry about the source being as required, then you can't.
> 

Not sure I understand your comment about "can't pass spaces in the flag list". Adding `rsyncopts=--chown user:group --stats` works fine in my simple test. Please provide a repro scenario for this.

> My initial thought for a simpler option, rather than "fixing" the parsing, was to maybe go with extra key options, e.g. rsyncopts, rsyncopts2 and rsynopts3 keys and tag them all onto the command?
> 
> Or maybe you'd prefer specific 'umask' and 'chown' keys directly handled by the plugin? Or specific 'chown' and 'chmod' keys which then get mapped to '--chmod' and '--chown' on the actual rsync command?
> 
> There's a bigger issue though with the default -a flag if specifying extra flags. I would argue that this needs to be removed if other flags are specified to stop clashes.
> 
> This paragraph started just by way of a note to maybe add somewhere, but is a good example of why the -a should be removable. If copying into the /etc/skel area then it seems to be necessary to override the default -a so that the files belong to root. Otherwise they don't seem to be copied when the user is created (at least I'm assuming that's why -> sequence of copydir '.config' as a user, copyfile '.bashrc' as root, create user and only the '.bashrc' is copied to the new user).
> 
> But, now I've tried with specifying -rpti, I still get the files belonging to the user number that they belonged to on the server that the burn was run on, rather than root. I can see that the -a is still present in the underlying rsync call.
> 

It could make sense for sdm to not add a gratuitous `-a` if `rsyncopts` is specified. Would that solve this for you?

> Something else I noticed before, the password for the created user (user created at burn time and password entered when prompted) being in the /etc/sdm/assets/user/redact-passwords file, even after the system has been setup and booted. I'm not sure that's a good thing.
> 

Presumably you used the `--redact` switch? If so, where did you apply it? If it's on the customize command, the file should have been deleted. Unfortunately, if you used it on the `--burn` command, a buglet prevents this from happening. Please identify which scenario you applied `--redact` to.

> Typo in the hotspot plugin "Configure hotspot with hostpad and dhcpcd" [sorry, ignore me :-)]
> 
Thx. 

> Using old style networking, I think there's extra config I need to do that I wouldn't on an off the shelf installation. Particularly that eth0 doesn't get setup. I didn't spot any instructions on this?
> 
What do you mean by "old style networking"? This might be worthy of a new issue so we can sort out what you're asking and resolve it.

> It seems that I need to enable / start the dhcpcd service and add some configuration for eth0. I'm still working on that (don't get much time to try any of this, especially without many interruptions!).
> 

Hmmm...why are you using dhcpcd? OK, useful to use dhcpcd on Bullseye, but on Buster, NM is really much better. I was skeptical at first, but at this point I am sold on it being a far superior solution to dhcpcd. Please clarify what you're trying to do. Maybe put this in the new issue with the above question about "old style networking"
> Thanks for all the great work.

Thanks! As someone said in the Pi forum, we hear a lot about the problems, but rarely get thanked for all the good stuff.

**Comment by stb68** [comment]
> > "user:group" still seems to not work and need to use "user.group" - although that now works without giving a warning.
> 
> Can you provide a specific example where this doesn't work? I of course tested it before heave-ho, so how can I repro this?

and

> Not sure I understand your comment about "can't pass spaces in the flag list". Adding rsyncopts=--chown user:group --stats works fine in my simple test. Please provide a repro scenario for this

I'll send you some output in a message showing these.

> It could make sense for sdm to not add a gratuitous -a if rsyncopts is specified. Would that solve this for you?
Yes, I'd say that's exactly what's needed.

> Presumably you used the --redact switch? If so, where did you apply it? If it's on the customize command, the file should have been deleted. Unfortunately, if you used it on the --burn command, a buglet prevents this from happening. Please identify which scenario you applied --redact to.

Actually I didn't include that at all!  Presumably I would need to add it to the burn though since I'm not creating the user until that point so sounds like it won't work anyway.

I'm not particularly worried about it and, even if I was, I know where the file is to remove it afterwards!

> What do you mean by "old style networking"? 

dhcpcd on Bookworm.  I'm quite happy with nw as the way forward, but need some time to get to grips with it first as I've never used it.  Also, it seems that the hotspot with it isn't as fully featured yet.  Plus, of course, I'll have to update my management scripts for things like switching IP ranges depending what network I'm connected to, switching channels, etc.  There's only so many hours in the day!

Basically, using my hacked hotspot plugin (I change the if raspiosver -lt 12 (or whatever the line was) to -lt 15 to let me select dhcpcd on Bookworm) and these lines in my @ file:
network:"netman=dhcpcd|noipv6"
my_hotspot:"type=routed|wlanip=192.168.254.1|hwmode=g|channel=10|device=wlan0|country=GB|dhcprange=192.168.254.20,192.168.4.64,255.255.255.0|leasetime=168h|passphrase=x1234|ssid=test234|hsmode=hostapd|enable=true"

the pi doesn't connect to the ethernet (one orange flashing light on the RJ45 connector IIRC).  On booting it complains that it can't get the time for 120 seconds before continuing, I assume because of this.  It's on my list to look into, not at all sure it's your issue.

Thanks (bed soon as I have a First Aid course all day tomorrow)
